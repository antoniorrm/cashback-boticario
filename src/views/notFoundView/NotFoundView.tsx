import React from 'react';
import {
  Box,
  Container,
  Typography,
  makeStyles
} from '@material-ui/core';

const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: theme.palette.background.default,
    paddingBottom: theme.spacing(3),
    paddingTop: theme.spacing(3)
  },
  image: {
    marginTop: 50,
    display: 'inline-block',
    maxWidth: '100%',
    width: 560
  }
}));

const NotFoundView = () => {
  const classes = useStyles();

  return (
    <div
      className={classes.root}
    >
      <Box
        display="flex"
        flexDirection="column"
        justifyContent="center"
      >
        <Container maxWidth="md">
          <Typography
            align="center"
            color="textPrimary"
            variant="h1"
          >
            404: Página não encontrada
          </Typography>
          <Typography
            align="center"
            color="textPrimary"
            variant="subtitle2"
          >
            Está página não está disponivel no sistema!
          </Typography>
        </Container>
      </Box>
    </div>
  );
};

export default NotFoundView;
