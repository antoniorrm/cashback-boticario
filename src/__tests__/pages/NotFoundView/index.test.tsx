/* eslint-disable no-undef */
import React from 'react'
import { render, screen } from '@testing-library/react'
import NotFoundView from '../../../pages/NotFoundView'

describe('NotFoundView component', () => {
	it('renders NotFoundView component correctly', () => {
		const { container } = render(<NotFoundView />)
		expect(container).toMatchSnapshot()
	})
	it('renders Texts in NotFoundView component correctly', () => {
		render(<NotFoundView />)
		const TextTitle = screen.getByText(/404: Página não encontrada/i)
		const Text = screen.getByText(/Está página não está disponivel no sistema!/i)
		expect(TextTitle).toBeInTheDocument()
		expect(Text).toBeInTheDocument()
	})
})
