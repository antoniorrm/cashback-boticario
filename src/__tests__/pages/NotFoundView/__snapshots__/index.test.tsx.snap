// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundView component renders NotFoundView component correctly 1`] = `
<div>
  <div
    class="makeStyles-root-1"
  >
    <div
      class="MuiBox-root MuiBox-root-3"
    >
      <div
        class="MuiContainer-root MuiContainer-maxWidthMd"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
        >
          404: Página não encontrada
        </h1>
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 MuiTypography-colorTextPrimary MuiTypography-alignCenter"
        >
          Está página não está disponivel no sistema!
        </h6>
      </div>
    </div>
  </div>
</div>
`;
